x-pack/legacy/plugins/ingest/server/rest_api/datasource/index.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import * as Joi from 'joi';
+import {
+  ReturnTypeCreate,
+  ReturnTypeUpdate,
+  ReturnTypeGet,
+  ReturnTypeList,
+} from '../../../common/types/std_return_format';
+import {
+  FrameworkRequest,
+  FrameworkRouteHandler,
+} from '../../libs/adapters/framework/adapter_types';
+import { ServerLibs, Datasource } from '../../libs/types';
+
+export const createGETDatasourceRoute = (libs: ServerLibs) => ({
+  method: 'GET',
+  path: '/api/ingest/datasource/{datasourceId}',
```

Comment:
nitpick: in fleet we uses plural for ressources name
```suggestion
path: '/api/ingest/datasources/{datasourceId}',
```

Reply:
I've updated singular to plural in all data source API paths, and did the same for policy APIs as well